src/contracts/libraries/GPv2Encoding.sol

```diff
@@ -110,52 +123,79 @@ library GPv2Encoding {
         );
         order.tip = abi.decode(encodedOrder[74:], (uint256));
         uint8 flags = uint8(encodedOrder[106]);
-        order.executedAmount = abi.decode(encodedOrder[107:], (uint256));
+        uint256 executedAmount = abi.decode(encodedOrder[107:], (uint256));
         uint8 v = uint8(encodedOrder[139]);
         bytes32 r = abi.decode(encodedOrder[140:], (bytes32));
         bytes32 s = abi.decode(encodedOrder[172:], (bytes32));
 
-        order.sellToken = tokens[order.sellTokenIndex];
-        order.buyToken = tokens[order.buyTokenIndex];
+        order.sellToken = tokens[sellTokenIndex];
+        order.buyToken = tokens[buyTokenIndex];
         order.kind = OrderKind((flags >> ORDER_KIND_BIT) & 0x1);
         order.partiallyFillable =
             (flags >> ORDER_PARTIALLY_FILLABLE_BIT) & 0x01 == 0x01;
 
         bytes32 orderDigest;
-        bytes32 signingDigest;
 
         // NOTE: In order to avoid a memory allocation per call by using the
         // built-in `abi.encode`, we reuse the memory region reserved by the
         // caller for the order result as input to compute the hash.
         // solhint-disable-next-line no-inline-assembly
         assembly {
-            // NOTE: Structs are not packed in Solidity, and there are 9 order
-            // fields to hash for a total of `9 * sizeof(uint) = 288` bytes.
-            orderDigest := keccak256(order, 288)
+            // NOTE: Structs are not packed in Solidity, and there is the order
+            // type hash, which is temporarily stored in the slot reserved for
```

Comment:
For me, this is a tad too hacky (temporarily reusing a field on a struct with some unrelated value). If I understand correctly the struct is reused for every order and as such would only allocate memory once per settlement. How much gas are we saving there? Is it really worth all this advanced assembly and extra code comments?

Reply:
This would perform an additional `abi.encode(ORDER_TYPE_HASH, order.sellToken, order....)` per call to decode order. This would allocate an additional 320 byte per `decodeOrder` call. Additionally, since this memory can't be freed, this would accumulate per order and not just be allocated once.